src/test/shell/bazel/path_mapping_test.sh

```diff
@@ -0,0 +1,290 @@
+    --noexperimental_worker_multiplex \
+    //src/main/java/com/example:Main &> $TEST_log || fail "run failed unexpectedly"
+  expect_log 'Hello, World!'
+  # JavaToolchainCompileBootClasspath, JavaToolchainCompileClasses and header compilation.
+  expect_log '3 \(linux\|darwin\|processwrapper\)-sandbox'
+  # Actual compilation actions.
+  expect_log '2 worker'
+  expect_not_log 'disk cache hit'
+
+  bazel run -c opt \
+    --disk_cache=$cache_dir \
+    --experimental_output_paths=strip \
+    --strategy=Javac=worker \
+    --worker_sandboxing \
+    --noexperimental_worker_multiplex \
+    //src/main/java/com/example:Main &> $TEST_log || fail "run failed unexpectedly"
+  expect_log 'Hello, World!'
+  expect_log '5 disk cache hit'
+  expect_not_log '[0-9] \(linux\|darwin\|processwrapper\)-sandbox'
+  expect_not_log '[0-9] worker'
+}
+
+function test_path_stripping_remote() {
+  bazel run -c fastbuild \
+    --experimental_output_paths=strip \
+    --remote_executor=grpc://localhost:${worker_port} \
+    //src/main/java/com/example:Main &> $TEST_log || fail "run failed unexpectedly"
+  expect_log 'Hello, World!'
+  # JavaToolchainCompileBootClasspath, JavaToolchainCompileClasses, 1x header compilation and 2x
+  # actual compilation.
+  expect_log '5 remote'
+  expect_not_log 'remote cache hit'
+
+  bazel run -c opt \
+    --experimental_output_paths=strip \
+    --remote_executor=grpc://localhost:${worker_port} \
+    //src/main/java/com/example:Main &> $TEST_log || fail "run failed unexpectedly"
+  expect_log 'Hello, World!'
+  expect_log '5 remote cache hit'
+  # Do not match "5 remote cache hit", which is expected.
+  expect_not_log '[0-9] remote[^ ]'
+}
+
+function test_path_stripping_remote_multiple_configs() {
+  mkdir rules
+  cat > rules/defs.bzl <<'EOF'
+LocationInfo = provider(fields = ["location"])
+
+def _location_setting_impl(ctx):
+    return LocationInfo(location = ctx.build_setting_value)
+
+location_setting = rule(
+    implementation = _location_setting_impl,
+    build_setting = config.string(),
+)
+
+def _location_transition_impl(settings, attr):
+    return {"//rules:location": attr.location}
+
+_location_transition = transition(
+    implementation = _location_transition_impl,
+    inputs = [],
+    outputs = ["//rules:location"],
+)
+
+def _bazelcon_greeting_impl(ctx):
+    content = """
+package com.example.{package};
+
+public class Lib {{
+  public static String getGreeting() {{
+    return String.format("Hello, BazelCon {location}!");
+  }}
+}}
+""".format(
+        package = ctx.attr.name,
+        location = ctx.attr.location,
+    )
+    file = ctx.actions.declare_file("Lib.java")
+    ctx.actions.write(file, content)
+    return [
+        DefaultInfo(files = depset([file])),
+    ]
+
+bazelcon_greeting = rule(
+    _bazelcon_greeting_impl,
+    cfg = _location_transition,
+    attrs = {
+        "_allowlist_function_transition": attr.label(default = "@bazel_tools//tools/allowlists/function_transition_allowlist"),
```

Comment:
Nit: do we still need this after https://github.com/bazelbuild/bazel/commit/bb7fb2d32f055f2a70a5ab394cf5aef29bc74b2e?

Reply:
We don't, removed.